homeassistant/components/template/light.py

```diff
@@ -613,6 +506,10 @@ async def async_turn_on(self, **kwargs: Any) -> None:  # noqa: C901
                 self._rgbw_color = None
             optimistic_set = True
 
+        return optimistic_set
+
+    def get_registered_scripts(self, **kwargs) -> Generator[tuple[str, dict]]:
```

Comment:
Why is this a generator?

Considering the structure is like this, will it ever yield more than a single script?
```
if a:
    yield x
elif b:
    yield y
else:
   yield z
```